packages/backend/src/functions/storage.ts

```diff
@@ -12,219 +12,263 @@ import { getSignedUrl } from "@aws-sdk/s3-request-presigner"
+                printLog(
+                    `The object associated w/ ${data.objectKey} key has been found in the ${data.bucketName} bucket`,
+                    LogLevel.LOG
+                )
 
                 return true
             }
         } catch (error: any) {
-            if (error.$metadata.httpStatusCode === 404 && !error.ETag) {
-                logMsg(`Object: ${data.objectKey} does not exist!`, MsgType.LOG)
+            if (error.$metadata.httpStatusCode === 403) logAndThrowError(SPECIFIC_ERRORS.SE_STORAGE_MISSING_PERMISSIONS)
 
-                return false
-            }
+            // @todo handle more specific errors here.
 
-            logMsg(`Generic error when checking for object on S3 bucket: ${error}`, MsgType.ERROR)
+            // nb. do not handle common errors! This method must return false if not found!
+            // const commonError = COMMON_ERRORS.CM_INVALID_REQUEST
+            // const additionalDetails = error.toString()
+
+            // logAndThrowError(makeError(
+            //     commonError.code,
+            //     commonError.message,
+            //     additionalDetails
+            // ))
         }
 
         return false
     }
 )
 
 /**
- * Generate a new AWS S3 pre signed url to upload/download an object (GET).
+ * Return a pre-signed url for a given object contained inside the provided AWS S3 bucket in order to perform a GET request.
+ * @notice the pre-signed url has a predefined expiration expressed in seconds inside the environment
+ * configuration of the `backend` package. The value should match the configuration of `phase2cli` package
+ * environment to avoid inconsistency between client request and CF.
  */
 export const generateGetObjectPreSignedUrl = functions.https.onCall(
-    async (data: any, context: functions.https.CallableContext): Promise<any> => {
-        if (!process.env.CONFIG_CEREMONY_BUCKET_POSTFIX) throw new Error(GENERIC_ERRORS.GENERR_WRONG_ENV_CONFIGURATION)
-        // requires auth
-        if (!context.auth) logMsg(GENERIC_ERRORS.GENERR_NO_AUTH_USER_FOUND, MsgType.ERROR)
+    async (data: BucketAndObjectKeyData, context: functions.https.CallableContext): Promise<any> => {
+        if (!context.auth) logAndThrowError(COMMON_ERRORS.CM_NOT_AUTHENTICATED)
 
-        if (!data.bucketName || !data.objectKey) logMsg(GENERIC_ERRORS.GENERR_MISSING_INPUT, MsgType.ERROR)
+        if (!data.bucketName || !data.objectKey) logAndThrowError(COMMON_ERRORS.CM_MISSING_OR_WRONG_INPUT_DATA)
 
-        // extract the bucket name and object key from the data
+        // Prepare input data.
         const { objectKey, bucketName } = data
 
-        // get the firestore database
+        // Get Firestore DB.
         const firestoreDatabase = admin.firestore()
 
-        // need to get the ceremony prefix from the bucket name
+        // Extract ceremony prefix from bucket name.
         const ceremonyPrefix = bucketName.replace(process.env.CONFIG_CEREMONY_BUCKET_POSTFIX!, "")
```

Comment:
Change to `process.env.AWS_CEREMONY_BUCKET_POSTFIX!`

Reply:
Wow, good catch!